R/source.R

```diff
@@ -67,6 +67,9 @@
 #' @export
 cpp_source <- function(file, code = NULL, env = parent.frame(), clean = TRUE, quiet = TRUE, cxx_std = Sys.getenv("CXX_STD", "CXX11"), dir = tempfile()) {
   stop_unless_installed(c("brio", "callr", "cli", "decor", "desc", "glue", "tibble", "vctrs"))
+  if (!missing(file) && !file.exists(file)) {
```

Comment:
`!missing(file)` seems a bit counter to our "house style". I wonder if the signature should become `file = NULL`, to indicate `file` is not required (at least, if `code` is provided)?

Reply:
Should this be informative about the name/path of the file that does not exist ?

Reply:
Yeah, I was thinking that too. I've added the non-existent `file` path to the error now.

Reply:
Thanks. Perhaps ultimately this is something `fs::` can help with ?